### DISCUSSION TITLE
Using 'Add_predictors_model' to add a "global" scale model into "local" scale model

### DISCUSSION THREAD

**Discussion by Daviser95** [iiasa/ibis.iSDM#120] [Q&A]
Hi @Martin-Jung!
I have a question about the function `add_predictors_model`, concerning the possibility of adding a model built on a larger scale to a model at a local scale. This would be needed to avoid some common problems (i.e., niche truncation). I am now working on models at a continental scale (Africa) and I would like to use these models (or the final ensemble model, but maybe it's not possible to add ensemble model as predictor model) as a predictor for a local scale model (in Namibia). If this function doesn't support the inclusion of models at different scales, I could crop the global-scale model and use the resulting cropped model as a simple predictor instead. I didn't try this process yet because currently I am working to obtain the global scale models. 

Moreover, I have noticed two problems using the function `validate `and `effects`. Specifically, using the function evaluate I never obtained a score for Continuous Boyce index which returns NA, while other indexes are computed. I tried both `validate(mod, method = "continuous")` and `validate(mod, method = "continuous", point = test1, point_column = "pres_abs")`. 
For the functions `effects`, I was able to obtain the graph reported in the reference site on GitHub (e.g., 'Train a basic model') only the first time, and then I got different graphs (for example, using xgboost engine) or strange errors. 

Lastly, I would very much support the creation of a function to implement different types of cross-validation during the modeling and validation phases. 

Thank you for your support!
Best whishes,
Davide

**Comment by Martin-Jung** [comment]
1) There is a misunderstanding, I think. The `add_predictors_model()` function adds the prediction outcome from an existing model to another one as predictor. It is not for ensemble modelling or similar. And yes, I would recommend to prepare your predictors specifically for each run (through clipping or cropping) rather than have the package do that.

2) `validate()` and here particular the CBI can fail for cases where you don't have enough presence/absence data. If you get an error, try to reproduce it with the testing data that come with the package and fill in an issue.

3) Cross-validation: Yes, there is an issue about that already ( #89 ), but it is not something that will be implemented (at least by me) in the package itself in the near future owing to time constraints and other things that have priority.

**Comment by Daviser95** [comment]
Thank you for these suggestions.
So basically it would be the same if I save the prediction from the continental-scale model, crop it, and add it as a predictor (`add_predictors`) to the local-scale model as a new variable?

**Reply by Martin-Jung** [reply]
Yes, that might give you also more control about it. 
So fit African-wide model, extract prediction (`fit$get_data()`), crop/mask via `terra` and then add to the 'local' model as predictor.
